[spm] add missing options

nxtstep opened the pull request:
🔑

Added `generate-xcodeproj`, `resolve` and `init` as possible `swift package` command options.

### Checklist
- [x] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [x] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [x] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### Motivation and Context
The existing `spm` action was missing commands that are available to use with `swift package *`. E.g. the non-trivial `generate-xcodeproj` was missing.

And aprospo I discovered a bug when using this action with xcpretty and a command in the package range.

### Description
Added `generate-xcodeproj`, `resolve` and `init` as `spm` command options.

And:
- Fixed a bug where the `spm` package commands would not execute correctly when xcpretty_output is used.

nxtstep commented:
I signed it!

janpio commented:
Your commit doesn't show your GitHub avatar @nxtstep, which probably means your GitHub account is registered with another email address than the one you used to create your git commits. You can either change your commits or add the commit email address to your CLA.

nxtstep committed:
Add resolve, generate-xcodeproj and init as possible `swift package` command options

And:
Fixed a bug where the package commands would not execute correctly when xcpretty_output is used

nxtstep commented:
@janpio Can we merge this one, or is there anything I can do?

nxtstep commented:
@joshdholtz could you pls 🙏 have a look at this PR. Much appreciated 🥇

Commit:
Updated authors as per @KrauseFx's request

nxtstep commented:
@joshdholtz @janpio @KrauseFx Is there anything holding you back from merging this one? If so, please let me know if I can help 🙏

joshdholtz reviewed (approved):
This looks solid ✊ Thanks for adding all those tests 🎊
